core/handler/manager_server.go

```diff
@@ -55,7 +58,8 @@ func (h *handlerManager) validateTTNAuthAppContext(ctx context.Context, appID st
 		}
 		token, err := h.handler.Component.ExchangeAppKeyForToken(appID, key)
 		if err != nil {
-			return ctx, nil, err
+			h.handler.Ctx.WithError(err).Warn("Could not exchange key for token")
+			return ctx, nil, errors.NewErrPermissionDenied("Could not exchange key for token")
```

Comment:
I think we should do this in the `ExchangeAppKeyForToken` func, as this error may either be caused by misconfiguration of the handler, or by an invalid access key. The first should be an InternalServerError, the second indeed a PermissionDenied